service/service.go

```diff
@@ -485,6 +486,34 @@ func (s *service) UpsertProductType(ctx context.Context, prodType *prodEntity.Pr
 	return nil
 }
 
+func (s *service) DeleteProductById(ctx context.Context, userId uuid.UUID, id uuid.UUID) error {
+	prodIds := []uuid.UUID{id}
+	prodImages, _, err := s.productRepository.GetProductImagesByProductIds(ctx, prodIds)
+	if err != nil {
+		return util.NewError(codes.Internal, errPb.StoreErrorCode_ERROR_WHEN_GETTING_PRODUCT_IMAGE.String(), "Error saat melakukan pencarian gambar produk : "+err.Error())
+	}
+
+	s.productRepository.InitiateTransaction(ctx)
+	removeProdImageIds := []string{}
+	for _, i := range prodImages {
+		log.Println("Remove : ", i.ID.String())
+		removeProdImageIds = append(removeProdImageIds, i.ID.String())
+	}
+	log.Printf("Remove product images : %v \n", removeProdImageIds)
+	if err := s.imageRepository.RemoveImage(ctx, removeProdImageIds, "product", userId.String()); err != nil {
+		s.productRepository.TransactionRollback()
+		return util.NewError(codes.Internal, errPb.StoreErrorCode_ERROR_WHEN_REMOVING_IMAGE_FROM_STORAGE.String(), "Error saat menghapus gambar dari penyimpanan : "+err.Error())
+	}
+	if err := s.productRepository.DeleteProductImages(ctx, prodImages); err != nil {
+		s.productRepository.TransactionRollback()
+		return util.NewError(codes.Internal, errPb.StoreErrorCode_ERROR_WHEN_DELETING_PRODUCT_IMAGE.String(), "Error saat menghapus gambar produk : "+err.Error())
+	}
+
+	err = s.productRepository.DeleteProductById(ctx, id)
```

Comment:
Please handle. if there is error, transaction will be rolledback

Reply:
Has been updated to handle error